### PR TITLE
nat: fix missing L3 pseudo-header csum diff applied into L4

### PR DESCRIPTION
**Pull request by sahid**
In context of an ICMP need to frag received, the NAT happening in the embedded packet is missing applying the L3 pseudo-header csum diff into the TCP/UDP header.

This commit is fixing the bug by reusing same pattern used in other rewrite_ functions.

### CODE REVIEW

**Comment by sahid** [comment]
o/ @jschwinger233, @ti-mo, @borkmann can you have a look? It's the following patch

File: bpf/lib/nat.h [hunk 1]
```diff
@@ -359,30 +359,30 @@ snat_v4_icmp_rewrite_ingress_embedded(struct __ctx_buff *ctx,
 				      struct ipv4_nat_entry *state,
 				      __u32 l4_off, __u32 inner_l4_off)
```
Comment (ldelossa) [inline bpf/lib/nat.h:360]: Not part of your PR, but when we get into this function `l4_off` is really `icmphdr_off`, right? If so might be a nice change while this code is being edited here.

**Review by ldelossa — APPROVED** [review]
Changes LGTM.

File: bpf/lib/nat.h [hunk 2]
```diff
@@ -421,6 +415,9 @@ snat_v4_icmp_rewrite_ingress_embedded(struct __ctx_buff *ctx,
 		return DROP_WRITE_ERROR;
 	if (ipv4_csum_update_by_diff(ctx, l4_off + sizeof(struct icmphdr), sum) < 0)
 		return DROP_CSUM_L3;
+	if (csum.offset &&
+	    csum_l4_replace(ctx, inner_l4_off, &csum, 0, sum, flags) < 0)
```
Comment (jschwinger233) [inline bpf/lib/nat.h:419]: In the case of UDP, this `csum_l4_replace` will run for twice. The first run is inside the `l4_modify_port` as you revised, the second run is here as `csum.offset` is non-zero.

File: bpf/lib/nat.h [hunk 3]
```diff
@@ -395,23 +395,17 @@ snat_v4_icmp_rewrite_ingress_embedded(struct __ctx_buff *ctx,
...
 			break;
-		}
-		default:
-			return DROP_UNKNOWN_L4;
-		}
-		/* By recomputing L4 checksum of inner packet we avoid having
-		 * to recompute L4 of the ICMP Error.
-		 */
-		suml4 = csum_diff(&tuple->dport, 4, &state->to_dport, 4, 0);
-		if (csum_l4_replace(ctx, inner_l4_off, &csum, 0, suml4, 0) < 0)
```
Comment (jschwinger233) [inline bpf/lib/nat.h:413]: I understand the bug you want to fix: the original `csum_l4_replace` wasn't called with flag `BPF_F_PSEUDO_HDR`. Then how about simply passing a `flag` variable to it?

**Pull request closed without merging**